### DISCUSSION TITLE
MONAI model/net zoo

### DISCUSSION THREAD

**Discussion by rgaiacs** [Project-MONAI/MONAI#2734] [General]
https://github.com/Project-MONAI/MONAI/tree/dev/monai/networks/nets is the home for MONAI model/net zoo. What are the rules for inclusion of new models/nets to the zoo?

**Comment by Nic-Ma** [comment]
Hi @rgaiacs ,

Thanks for your interest here.
If you are interested to contribute some new models/nets, that's great, welcome to submit a PR!
And I think the basic rule is to support both 2D and 3D data, and leverage the existing common blocks and layers to simplify the implementation and unify the APIs.

Thanks.

**Reply by lisun-ai** [reply]
Thanks a lot for your reply. I wonder if it's possible to contribute a model that only supports 3D data?

**Reply by Nic-Ma** [reply]
I think most of our common layers and blocks support both 2D and 3D, so you can simply leverage them to build your network.
CC @wyli @ericspod for more discussion here.

Thanks.

**Reply by wyli** [reply]
Yes we aim to support at least both 2d and 3d. You can alternatively submit it to https://github.com/Project-MONAI/research-contributions first.

**Reply by ericspod** [reply]
Anything novel and general purpose that's proven should go into core as actual network definitions. If the network is based off a paper with a specific implementation but the MONAI version is more general, somehow we'd want to include both the "reference" network and the general network.
